cmd/ipfs/main.go

```diff
@@ -315,6 +315,13 @@ func callCommand(ctx context.Context, req cmds.Request, root *cmds.Command) (cmd
 		return nil, err
 	}
 
+	if cmd.PreRun != nil {
+		err = cmd.PreRun(req)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
@mappum cant we generate the response a tiny bit earlier? pass it into `root.Call` ? (or call PreRun + PostRun from call?)